### ISSUE TITLE
Ping Loop Event not getting cleared on reconnect

### ISSUE THREAD

**Issue by harmon** [miguelgrinberg/python-engineio#153]
I can't quite put my finger on the exact issue, but we are getting an unexpected disconnect after a client.py reconnect because I think the ping_loop_event is no longer getting cleared with a recent code change in v3.11.0:

https://github.com/miguelgrinberg/python-engineio/commit/e38daad301d4b855cabf6a289b7348a58a314273

Was this `self.ping_loop_event.clear()` supposed to be removed?

```
09:41:12	- INFO     - [123145481080832 socketio] - engineio.client - 	WebSocket connection accepted with {'sid': 'f27467c12f3c4d85a2925bb4e9f1653f', 'upgrades': [], 'pingTimeout': 60000, 'pingInterval': 25000}
09:41:12	- INFO     - [123145481080832 socketio] - socketio.client - 	Engine.IO connection established
09:41:12	- INFO     - [123145465315328 Thread-5] - engineio.client - 	Sending packet PING data None
09:41:12	- INFO     - [123145465315328 Thread-5] - engineio.client - 	PONG response has not been received, aborting
09:41:12	- INFO     - [123145465315328 Thread-5] - engineio.client - 	Exiting ping task
09:41:12	- WARNING  - [123145470570496 Thread-6] - engineio.client - 	WebSocket connection was closed, aborting
```

I realize I don't have much detail in this ticket, but if I get time this week I can flush it out with more info on how to recreate.

**Comment by harmon** [comment]
Ok, I guess that's fine, since the default state of a new Event is already "cleared()", something else about that version bump must be causing my issues. Still looking.

**Comment by harmon** [comment]
Ahh, yes! Ok, figured it out: On reconnect, the `self.ping_loop_event` already exists, and does not get cleared again, so it basically disconnects again immediately. 

The change from:

```
    def _ping_loop(self):
        """This background task sends a PING to the server at the requested
        interval.
        """
        self.pong_received = True
        self.ping_loop_event.clear()
```
To:
```
    def _ping_loop(self):
        """This background task sends a PING to the server at the requested
        interval.
        """
        self.pong_received = True

        if self.ping_loop_event is None:
            self.ping_loop_event = self.create_event()
```

Means that the second time a _ping_loop thread is created, it isn't reset.

Maybe it needs:

```
    def _ping_loop(self):
        """This background task sends a PING to the server at the requested
        interval.
        """
        self.pong_received = True

        if self.ping_loop_event is None:
            self.ping_loop_event = self.create_event()
        else:
            self.ping_loop_event.clear()
```

**Comment by miguelgrinberg** [comment]
Yes, I messed this up with that change. I just put a fix in master. Would you like to give it a try before I cut a new release?

**Comment by harmon** [comment]
I tested the fix in my own file, so I think it's good.

**Comment by harmon** [comment]
I just merged a requirements lockdown to 3.10.0 to our code, and then you fixed it! 

Such appreciation, much service!
